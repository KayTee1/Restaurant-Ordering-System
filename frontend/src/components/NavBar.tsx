import { Button, Container, Nav, Navbar as NavbarBs } from "react-bootstrap";
import { NavLink } from "react-router-dom";

import axios from 'axios';

const handleResetOrders = async () => {
  try {
    const resetUrl = "http://localhost:5000/api/orders";
    const res = await axios.delete(resetUrl);
    console.log(res);
  } catch (err) {
    console.error(err);
  }
}

const NavBar = () => {
  return (
    <>
      <NavbarBs className="bg-white shadow-sm mb-3">
        <Container>
          <Nav className="me-auto">
            <Nav.Link as={NavLink} to={"/"}>
              Menu
            </Nav.Link>
            <Nav.Link as={NavLink} to={"/cart"}>
              Cart
            </Nav.Link>
            <Nav.Link as={NavLink} to={"/checkout"}>
              Checkout
            </Nav.Link>
          </Nav>
          <Button style={{ opacity: 0 }} onClick={handleResetOrders}>
            Reset Orders
          </Button>
        </Container>
      </NavbarBs>
    </>
  );
};

export default NavBar;
