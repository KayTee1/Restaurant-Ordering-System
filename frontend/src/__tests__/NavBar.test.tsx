import { expect, it, describe } from "vitest";
import { MemoryRouter } from "react-router-dom";
import { createRenderer } from "react-test-renderer/shallow";

import NavBar from "../components/NavBar";

const renderer = createRenderer();

const defaultComponent = (
  <MemoryRouter initialEntries={["/"]}>
    <NavBar />
  </MemoryRouter>
);

// Tests
describe("On the NavBar component", () => {
  it("should render and match the snapshot", () => {
    renderer.render(defaultComponent);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("should render the links", () => {
    renderer.render(defaultComponent);
    const h1 = renderer.getRenderOutput();
    expect(h1).toBeTruthy();
  });
});
