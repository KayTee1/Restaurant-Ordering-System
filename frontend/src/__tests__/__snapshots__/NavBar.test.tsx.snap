// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`On the NavBar component > should render and match the snapshot 1`] = `
<Router
  location={
    {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    {
      "action": "POP",
      "createHref": [Function],
      "createURL": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "index": 0,
      "listen": [Function],
      "location": {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NavBar />
</Router>
`;
